### PR TITLE
Refactor: deploy handling

### PR DESCRIPTION
**Pull request by till**
Related: DEVOPS-119

### CODE REVIEW

**Comment by till** [comment]
@fh tested this on InfoLit Playground. IMO we better wait until the current stable merge is done. But yeah, this is what I would refactor it to. Could probably use a few more tests.

**Comment by fh** [comment]
@till Needs rebase - if you want to, maybe also some tests?

**Comment by fh** [comment]
@till Shall I merge this now? Stable has been updated, but I remember you talking about tests you are currently building yesterday :)

**Comment by till** [comment]
@fh yeah, I will dig around with the tests a few more times today and tomorrow.

File: easybib/libraries/deploy.rb [hunk 1]
```diff
@@ -63,20 +63,19 @@ def allow_deploy(application, requested_application, requested_role = nil, node
     end
 
     def is_app_configured_for_stack(application, requested_application, requested_role, instance_roles)
-      case application
-      when requested_application
+      if application == requested_application
```
Comment (till) [inline easybib/libraries/deploy.rb]: I refactored this when I "read" the code in order to build a fixture for `node['opsworks']`.

cc @fh

File: easybib/spec/fixtures/metadata.rb [hunk 2]
```diff
@@ -2,3 +2,5 @@
 version '0.99.0'
 
 depends 'easybib'
+depends 'nginx-app'
```
Comment (till) [inline easybib/spec/fixtures/metadata.rb]: Needed to step through all the way into `easybib_nginx`.

**Pull request merged**